src/graph_notebook/magics/graph_magic.py

```diff
@@ -337,40 +378,89 @@ def gremlin(self, line, cell, local_ns: dict = None):
+
         if mode == QueryMode.EXPLAIN:
             res = self.client.gremlin_explain(cell)
             res.raise_for_status()
+            gremlin_metadata = set_request_metrics(metadata=gremlin_metadata,
+                                                   time_elapsed=res.elapsed, status=res.status_code,
+                                                   status_ok=res.ok, content=res.content)
             query_res = res.content.decode('utf-8')
+            predicates_regex = re.search(r'# of predicates: (.*?)\n', query_res)
+            if predicates_regex:
+                gremlin_metadata.set_metric_value('predicates', int(predicates_regex.group(1)))
+            explain_output = widgets.Output(layout=DEFAULT_LAYOUT)
+            children.append(explain_output)
+            titles.append('Explain')
             if 'Neptune Gremlin Explain' in query_res:
                 html = pre_container_template.render(content=query_res)
             else:
                 html = pre_container_template.render(content='No explain found')
-
-            explain_output = widgets.Output(layout=DEFAULT_LAYOUT)
             with explain_output:
                 display(HTML(html))
-            tab.children = [explain_output]
-            tab.set_title(0, 'Explain')
-            display(tab)
         elif mode == QueryMode.PROFILE:
             res = self.client.gremlin_profile(cell)
             res.raise_for_status()
+            gremlin_metadata = set_request_metrics(metadata=gremlin_metadata,
+                                                   time_elapsed=res.elapsed, status=res.status_code,
+                                                   status_ok=res.ok, content=res.content)
             query_res = res.content.decode('utf-8')
+
+            # Query execution time in Neptune DB
+            querytime_regex = re.search(r'Query Execution: (.*?)\n', query_res)
+            if querytime_regex:
+                gremlin_metadata.set_metric_value('query_time', float(querytime_regex.group(1)))
+            # # of predicates
+            predicates_regex = re.search(r'# of predicates: (.*?)\n', query_res)
+            if predicates_regex:
+                gremlin_metadata.set_metric_value('predicates', int(predicates_regex.group(1)))
+            # Count of results
+            count_regex = re.search(r'Count: (.*?)\n', query_res)
+            if count_regex:
+                gremlin_metadata.set_metric_value('results', int(count_regex.group(1)))
+            # Serialization
+            serialization_regex = re.search(r'Serialization: (.*?)\n', query_res)
+            if serialization_regex:
+                gremlin_metadata.set_metric_value('seri_time', float(serialization_regex.group(1)))
+            # Size of results returned
+            results_size_regex = re.search(r'Response size (bytes): (.*?)\n', query_res)
+            if results_size_regex:
+                gremlin_metadata.set_metric_value('results_size', int(results_size_regex.group(1)))
```

Comment:
Do you think we can move this to its own helper method and then have a test written against it? I'd like to make sure that these regexes have some coverage on them and currently we don't have a good way to do that for things that are inside our magic commands.

Reply:
Done, moved to set_gremlin_profile_metrics method under Metadata class, and unit test added.